### PR TITLE
Accept floats for radius km and mi in within param

### PR DESCRIPTION
**Pull request by Krysis7**
Event endpoint param `within` now allows floats for radius `km` and `mi`.

This pull request updates the SDK to support the new behaviour.

### CODE REVIEW

File: tests/endpoints/test_schemas.py [hunk 1]
```diff
@@ -163,37 +156,12 @@ class SchemaExample(schemas.Model):
...
     assert m.import_data(
         {"around": {"origin": "40.730610,-73.935242", "offset": "1km", "scale": "2km", "decay": "0.1"}}
-    ).to_primitive() == {"around": {"origin": u"40.730610,-73.935242", "decay": 0.1, "scale": u"2km", "offset": u"1km"}}
+    ).to_primitive() == {"around": {"origin": "40.730610,-73.935242", "decay": 0.1, "scale": "2km", "offset": "1km"}}
 
     with pytest.raises(schemas.SchematicsDataError):
         m.import_data({"around": "40.730610,-73.935242"}, validate=True)
 
 
-def test_area_model():
```
Comment (mbonnefoy) [inline tests/endpoints/test_schemas.py:172]: Is this still relevant to remove that test?

Comment (Krysis7) [reply tests/endpoints/test_schemas.py:172]: Updated and brought back the test since we are using Area class now.

File: predicthq/endpoints/schemas.py [hunk 2]
```diff
@@ -126,10 +124,10 @@ def _normalize(self, value):
 
 class Area(StringModel):
 
-    import_format = r"(?P<radius>\d+(k?m|ft|mi))@(?P<latitude>-?\d+(\.\d+)?),(?P<longitude>-?\d+(\.\d+)?)"
+    import_format = r"(?P<radius>(\d+(\.\d+)?)(km|mi)|(\d+)(m|ft))@(?P<latitude>[\-\+]?\d+(\.\d+)?),(?P<longitude>[\-\+]?\d+(\.\d+)?)"
     export_format = "{radius}@{latitude},{longitude}"
 
-    radius = StringType(regex=r"\d+(k?m|ft|mi)", required=True)
+    radius = StringType(regex=r"(\d+(\.\d+)?)+(km|mi)|(\d+)+(m|ft)", required=True)
```
Comment (mbonnefoy) [inline predicthq/endpoints/schemas.py:130]: You've got some extra `+` here?

Comment (mbonnefoy) [reply predicthq/endpoints/schemas.py:130]: I am not sure that you got your parentheses right.

File: predicthq/endpoints/v1/events/schemas.py [hunk 3]
```diff
@@ -26,6 +26,7 @@
     StringListType,
     StringModelType,
     StringType,
+
```
Comment (mbonnefoy) [inline predicthq/endpoints/v1/events/schemas.py:29]: This extra newline is not needed

File: tests/endpoints/test_schemas.py [hunk 4]
```diff
@@ -85,33 +85,26 @@ class SchemaExample(schemas.Model):
 def test_string_list_type():
     class SchemaExample(schemas.Model):
 
-        area_list = schemas.StringListType(schemas.StringModelType(schemas.Area), separator="+")
         string_list = schemas.StringListType(schemas.StringType, separator="+")
 
-    string_data = {"string_list": "a+b+c", "area_list": "10km@-36.847585,174.765742+10km@-41.288058,174.778265"}
+    string_data = {"string_list": "a+b+c"}
```
Comment (mbonnefoy) [inline tests/endpoints/test_schemas.py:90]: Any specific reason for removing the list? Applies below as well.

File: tests/endpoints/test_schemas.py [hunk 5]
```diff
@@ -163,23 +156,23 @@ class SchemaExample(schemas.Model):
...
     with pytest.raises(schemas.SchematicsDataError):
         m.import_data({"around": "40.730610,-73.935242"}, validate=True)
 
 
-def test_area_model():
+def test_location_model():
     class SchemaExample(schemas.Model):
 
-        area = schemas.StringModelType(schemas.Area)
+        location = schemas.StringModelType(schemas.Location)
```
Comment (mbonnefoy) [inline tests/endpoints/test_schemas.py:168]: Looks like it got things a bit odd because you just switched the order? Might be better to keep the previous to make sure that you didn't lose anything in the change?

**Review by mbonnefoy — CHANGES_REQUESTED** [review]

File: predicthq/endpoints/schemas.py (hunk 2 above)
Comment (Krysis7) [reply predicthq/endpoints/schemas.py:130]: ok, it doesnt seme to affects the actual regex after trying it a bit, but it better to wrap it just incase.

File: tests/endpoints/test_schemas.py (hunk 4 above)
Comment (Krysis7) [reply tests/endpoints/test_schemas.py:90]: yep it was due to me removing the area class will bring back the previous area test cases.

File: tests/endpoints/test_schemas.py (hunk 5 above)
Comment (Krysis7) [reply tests/endpoints/test_schemas.py:168]: yep it was due to me removing the area class will bring back the previous area test cases.

File: predicthq/endpoints/v1/events/schemas.py (hunk 3 above)
Comment (Krysis7) [reply predicthq/endpoints/v1/events/schemas.py]: removed

File: predicthq/endpoints/schemas.py [hunk 6]
```diff
@@ -126,10 +124,10 @@ def _normalize(self, value):
 
 class Area(StringModel):
 
-    import_format = r"(?P<radius>\d+(k?m|ft|mi))@(?P<latitude>-?\d+(\.\d+)?),(?P<longitude>-?\d+(\.\d+)?)"
+    import_format = r"(?P<radius>(\d+(\.\d+)?)(km|mi)|(\d+)(m|ft))@(?P<latitude>[\-\+]?\d+(\.\d+)?),(?P<longitude>[\-\+]?\d+(\.\d+)?)"
     export_format = "{radius}@{latitude},{longitude}"
 
-    radius = StringType(regex=r"\d+(k?m|ft|mi)", required=True)
+    radius = StringType(regex=r"((\d+(\.\d+)?)+(km|mi)|(\d+)+(m|ft))", required=True)
```
Comment (mbonnefoy) [inline predicthq/endpoints/schemas.py:130]: You've got the extra `+` here as well?

File: tests/endpoints/test_schemas.py [hunk 7]
```diff
@@ -270,4 +270,4 @@ def load_page(self, page):
     assert list(p1.iter_all()) == list(p1) + list(p2) + list(p3)
 
     for item in p1.iter_all():
-        assert item._endpoint == endpoint
+        assert item._endpoint == endpoint
```
Comment (mbonnefoy) [inline tests/endpoints/test_schemas.py:273]: Minor: you've got a missing newline here.

**Review by mbonnefoy — CHANGES_REQUESTED** [review]

**Review by mbonnefoy — APPROVED** [review]
